src/components/Jobs/MonitorJobs/MonitorJobs.js

```diff
@@ -229,7 +231,17 @@ const MonitorJobs = ({
     jobRun => {
       return enrichRunWithFunctionTag(jobRun, fetchJobFunctions, fetchJobFunctionsPromiseRef).then(
         jobRun => {
-          setSelectedJob(jobRun)
+          fetchJobFunctionsPromiseRef.current.then(func => {
+            if (func && func[0].spec) {
+              jobRun = {
+                ...jobRun,
+                runOnSpot: capitalize(func[0].spec.preemption_mode ?? ''),
+                nodeSelectorChips: parseKeyValues(func[0].spec.node_selector || {}),
+                priority: generateFunctionPriorityLabel(func[0].spec.priority_class_name ?? '')
+              }
+            }
+            setSelectedJob(jobRun)
+          })
```

Comment:
we already use the function `enrichRunWithFunctionTag` to add additional properties to the run from the function. You can more this logic there and rename the function.